api/all/src/main/java/io/opentelemetry/api/internal/ConfigUtil.java

```diff
@@ -19,6 +21,17 @@ public final class ConfigUtil {
 
   private ConfigUtil() {}
 
+  /**
+   * Returns a copy of system properties which is safe to iterate over.
+   *
+   * <p>In java 8 and android environments, iterating through system properties may trigger {@link
+   * ConcurrentModificationException}. This method ensures callers can iterate safely without risk
+   * of exception. See https://github.com/open-telemetry/opentelemetry-java/issues/6732 for details.
+   */
+  public static Properties safeSystemProperties() {
+    return (Properties) System.getProperties().clone();
```

Comment:
Cloning seems to be the simplest way to accommodate various use cases. Anything other option that returns `Properties` ends up mimicking the logic of `clone()` anyway. If `clone()` ends up being too expensive (I doubt it), we can cache a copy.

Reply:
I'm ok with it

`ConfigUtil.getString(..)` is already slow (too slow to be used on the hotpath) since it normalizes all the entry keys on each call

Reply:
>  If clone() ends up being too expensive

`clone` is only needed on jdk8 and android so we could skip it on other jdks if we so wish. Untested code follows
```java
private static final CLONE_PROPERTIES = System.getProperty("java.version").equals("0") // android
  || System.getProperty("java.specification.version").startsWith("1."); // jdk8
```

Reply:
> clone is only needed on jdk8 and android so we could skip it on other jdks if we so wish.

That's true. I'd prefer to keep that in our back pocket in case we need it. As a general rule, I think we avoid runtime-specific code unless we have a strong reason (subjective definition of "strong").